src/trio/_path.py

```diff
@@ -32,219 +22,102 @@
         OpenBinaryModeWriting,
         OpenTextMode,
     )
-    from typing_extensions import Concatenate, Literal, ParamSpec, TypeAlias
-
-    from trio._file_io import AsyncIOWrapper as _AsyncIOWrapper
+    from typing_extensions import Concatenate, Literal, ParamSpec
 
     P = ParamSpec("P")
 
-T = TypeVar("T")
-StrPath: TypeAlias = Union[str, "os.PathLike[str]"]  # Only subscriptable in 3.9+
-
+    PathT = TypeVar("PathT", bound="Path")
+    T = TypeVar("T")
 
-# re-wrap return value from methods that return new instances of pathlib.Path
-def rewrap_path(value: T) -> T | Path:
-    if isinstance(value, pathlib.Path):
-        return Path(value)
-    else:
-        return value
 
+def _wraps_async(
+    wrapped: Callable[..., Any]
+) -> Callable[[Callable[P, T]], Callable[P, Awaitable[T]]]:
```

Comment:
FYI, `pyright` tracks `P` through and attaches the docstring from the initial point to where it ends up, so that's a potential solution for getting `pyright` to recognize the docstrings.

Reply:
I don't understand. What's `pyright` doing with docstrings and paramspecs? And what does that have to do with inconsistent types?

If you have a solution to the `pyright` problem I'd love to know more! ❤️

Reply:
pyright --verifytypes checks for docstrings, and while it doesn't highlight with an error message for each one you can see it in the summary at the end:
```
  Symbols without documentation:
    Functions without docstring: 86
    Functions without default param: 0
    Classes without docstring: 21
```
but that was broken before (see #2910) so not a hard requirement to get it passing. Although #2910 says 133 functions are missing docstrings, and here it's 86, so you maybe you did fix the ones from path? Or it's a matter of `--ignoreexternal`??

Reply:
I suspect that inheriting a bunch of methods from `PurePath` rather than wrapping them has indeed fixed the docstrings! 💪